R/workflow.R

```diff
@@ -50,10 +54,23 @@ workflow <- R6::R6Class(
                                       workflow_path)
     },
 
-    run = function() {
-      ## TODO: Implement run
-      orderly_log("workflow", paste0("Running workflow ", self$workflow_id))
-      stop("Workflow running not yet implemented")
+    run = function(envir = NULL, message = NULL, instance = NULL,
+                   remote = NULL) {
+      orderly_log("workflow", sprintf("Running workflow '%s' with ID '%s'",
+                                      self$workflow_name, self$workflow_id))
+      output <- lapply(self$steps, run_single, envir, self$config, message,
+                       instance, remote)
+      run_ids <- unlist(lapply(output, function(out) {
+        out$value
+      }))
+      out_message <- sprintf("Completed running workflow '%s' with ID '%s'\n",
+                             self$workflow_name, self$workflow_id)
+      no_failed <- length(self$steps) - length(run_ids)
+      if (no_failed > 0) {
+        out_message <- sprintf("%s with %s failure(s)", out_message, no_failed)
+      }
+      orderly_log("workflow", out_message)
```

Comment:
If you look at `orderly_log_style` then you can get a custom colour on the workflow, which I think would be useful

Reply:
You might also want to do this last one as a multiline message (if the 2nd arg to `orderly_log` is a character vector you get nice printing over multiple lines) with `<name>:<id>` pairs?

Reply:
When we add cli support for this we should wire up the notifications too

Reply:
Made these print green now. The style is just "workflow" which isn't very generic unlike the others (highlight, alert) so happy to change that if you would prefer something more general?